events/adapters.go

```diff
@@ -0,0 +1,34 @@
+package events
+
+import (
+	"errors"
+
+	"github.com/flanksource/incident-commander/api"
+	"github.com/flanksource/postq"
+)
+
+func AsyncAdapter(fn func(api.Context, postq.Events) postq.Events) postq.AsyncEventHandlerFunc {
+	return func(ctx postq.Context, events postq.Events) postq.Events {
+		c, ok := ctx.(api.Context)
+		if !ok {
+			for i := range events {
+				events[i].SetError("invalid context")
+			}
+
+			return events
+		}
+
+		return fn(c, events)
+	}
+}
+
+func SyncAdapter(fn func(api.Context, postq.Event) error) postq.SyncEventHandlerFunc {
+	return func(ctx postq.Context, event postq.Event) error {
+		c, ok := ctx.(api.Context)
+		if !ok {
+			return errors.New("invalid context")
+		}
+
+		return fn(c, event)
+	}
+}
```

Comment:
@moshloop Due to the way we're accepting event handler functions, I wasn't able to simply pass in our existing  funcs because those funcs were using `api.Context` whereas the sync/async handler requires the func to accept `postq.Context`

**Required**

```
func(postq.Context, postq.Event) error
```

**Have:**

```
func(api.Context, postq.Event) error
```

Even though our `api.Context` implements `postq.Context`, Go sees it as 2 different types.

I'm not really liking this adapter pattern so let me know if you have better ideas.

cc @yashmehrotra

Reply:
Can we not use generics or a helper func for this e.g. `postq.SyncHandlers(addNotificationEvent, schedulePlaybookRun)`